Xenakios/TakeRenaming.cpp

```diff
@@ -114,149 +115,84 @@ void AddToRenameLog(string &oldfilename,string &newfilename)
 #endif
 }
 
-void DoRenameSourceFileDialog666(COMMAND_T* ct)
+// ct->user==0: rename source only, ==1: rename source and take
+void DoRenameTakeAndSourceFileDialog(COMMAND_T* ct)
 {
 	vector<MediaItem_Take*> thetakes;
 	vector<MediaItem_Take*> alltakes;
 	XenGetProjectTakes(alltakes,false,false);
 	XenGetProjectTakes(thetakes,true,true);
 	if (thetakes.size()==0) return;
-	g_renameparams.takesToRename=(int)thetakes.size();
-	g_renameparams.mode=1;
-	bool bChanges = false;
-	for (int i=0;i<(int)thetakes.size();i++)
+	g_renameparams.takesToRename=thetakes.size();
+	g_renameparams.mode=2;
+	for (size_t i=0;i<thetakes.size();i++)
 	{
 		PCM_source *thesrc=(PCM_source*)GetSetMediaItemTakeInfo(thetakes[i],"P_SOURCE",0);
-		if (thesrc && strcmp(thesrc->GetType(),"SECTION")!=0 && thesrc->GetFileName() && thesrc->GetFileName()[0])
-		{
-			g_renameparams.curTakeInx=i+1;
-			string oldname;
-			oldname.assign(thesrc->GetFileName());
-			vector<string> fnsplit;
-			SplitFileNameComponents(oldname,fnsplit);
-			g_renameparams.OldName.assign(fnsplit[1]);
+		// if only renaming source
+		// skip takes which don't have a 'real' source
+		if (!thesrc || (ct->user == 0 && !SWS_GetSourceFileName(thesrc)))
```

Comment:
"Xenakios/SWS: Rename take source files... (no undo)" used to do nothing on MIDI items. Such sources have an empty filename, which is not skipped anymore.

```suggestion
		if (!thesrc || (ct->user == 0 && (!SWS_GetSourceFileName(thesrc) || !SWS_GetSourceFileName(thesrc)[0])))
```